js/reference.js

```diff
@@ -4215,6 +4215,51 @@
             return this._uniqueId;
         },
 
+        get citation() {
+            if (this._citation === undefined) {
+                var table = this._pageRef.table;
+                if (table.annotations.contains(module._annotations.CITATION)) {
+                    /**
+                     * citation specific properties include:
+                     *   - journal*
+                     *   - author
+                     *   - title
+                     *   - year*
+                     *   - url*
+                     *   - id
+                     * other properties:
+                     *   - template_engine
+                     */
+                    var citationAnno = table.annotations.get(module._annotations.CITATION).content;
+
+                    // if required fields are present, render each template and set that value on the citation object
+                    if (citationAnno.journal_pattern && citationAnno.year_pattern && citationAnno.url_pattern) {
+                        var options = {
+                            templateEngine: citationAnno.template_engine // if undefined, _renderTemplate defaults to Mustache
+                        };
+                        var keyValues = module._getFormattedKeyValues(table, this.reference._context, this._data, this._linkedData);
```

Comment:
Change `this.reference._context` to `this._pageRef._context`. The `reference` is a new reference that we're creating just for the tuple.